### ISSUE TITLE
Lint `unused_braces` is incorrectly identifying unused braces in a constant expression in a type using const generics

### ISSUE THREAD

**Issue by raphaelcohn** [rust-lang/rust#85744]
I tried this code:

```rust
pub struct PhysicalIdentifier(ArrayVec<u8, {Self::MaximumLength}>);

impl PhysicalIdentifier
{
	/// Maximum length.
	pub const MaximumLength: usize = MAX_PHYS_ITEM_ID_LEN;
}
```

This compiles as expected, but the lint `unused_braces` issues a warning:-

```
warning: unnecessary braces around block return value
 --> linux-support/src/network_device/PhysicalIdentifier.rs:9:44
  |
9 | pub struct PhysicalIdentifier(ArrayVec<u8, {Self::MaximumLength}>);
  |                                            ^^^^^^^^^^^^^^^^^^^^^ help: remove these braces
```

Removing the braces causes a compilation failure:-

```
error[E0747]: type provided when a constant was expected
 --> linux-support/src/network_device/PhysicalIdentifier.rs:9:44
  |
9 | pub struct PhysicalIdentifier(ArrayVec<u8, Self::MaximumLength>);
  |                                            ^^^^^^^^^^^^^^^^^^^
  |
help: if this generic argument was intended as a const parameter, surround it with braces
  |
9 | pub struct PhysicalIdentifier(ArrayVec<u8, { Self::MaximumLength }>);
```

Personally, I think the lint is on to something - putting braces around `Self::MaximumLength` is messy - but as things stand, it's a bug.

**Comment by SNCPlay42** [comment]
I can't reproduce with [this](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=eb74b609a50edf6c028d7b7b3ddb9f1c) code on playground.

**Comment by raphaelcohn** [comment]
@SNCPlay42 Original full code example is in version 0.0.23 of linux-support, running against rust 1.51; eg <https://crates.io/crates/linux-support>.

**Comment by qiujiangkun** [comment]
#85881
Similar issue

**Comment by JohnTitor** [comment]
I cannot reproduce it as well, I guess you were using an old nightly that is buggy. If it's still reproducible with the latest nightly, please re-open with the nightly version and code, thanks!
